docs/download.md

```diff
@@ -0,0 +1,122 @@
+---
+layout: page
+title: Download
+nav_order: 10
+---
+
+## Release v0.3.0
+This release includes additional performance improvements, for example, use of per thread default
+stream to make more efficient use of the GPU.  There is also further support for adaptive query
+execution, with more rewritten query plans now able to run on the GPU.  There are performance
+improvements for reading small Parquet files.  RAPIDS Shuffle with UCX has been updated to UCX
+1.9.0.
+
+Additional functionality includes Parquet reading for lists and structs, lead/lag for windows, and
+greatest/least operators among others.
+
+The release is supported on Apache Spark 3.0.0, 3.0.1, Databricks 7.3 ML LTS and Google Cloud
+Platform Dataproc 2.0.
+
+GPU reading of cached dataframes is enabled but only effective for Spark 3.1, for those who
```

Comment:
This requires the user configure the batched cache serializer, correct?  If we're going to mention it here we may want to point them to the documentation on how to do this.  AFAIK it won't "just work" if they use Spark 3.1 with the plugin.